### DISCUSSION TITLE
Instructions for Clone and Build locally?

### DISCUSSION THREAD

**Discussion by salderma** [gusztavvargadr/packer#449] [Q&A] [answered]
Hello,

First thank you for your work on these images.  I have utilized them for various vagrant proof-of-concept projects for several years.

The organization I'm now working for has set security policy such that they are not a fan of utilizing container and vm images that are sourced from the public.  As a result, I have cloned your project and am attempting to perform some manual builds locally on my workstation (first) and then hope to convert your AzureCI process over to be used with our Gitlab and Artifactory platforms - where we have multiple levels of security scanning infrastructure built.

I saw there's an issue still open dating back to 2016 on this topic (or at least it seems to be on this topic) https://github.com/gusztavvargadr/packer/issues/14.  Since that time, I'm sure the build process has changed.

Would anyone be able to describe the build process in a way that can be accomplished on a linux desktop here?  If I can reproduce the process, I would be willing to formally write it into a markdown document and submit a PR to add it to this project after I sort out the details. :beers:

**Comment by gusztavvargadr** [comment, marked as answer]
@salderma thanks for reaching out. It is a very valid approach to not to trust public images, that's also why I make the sources and building them available as well.

Regarding the documentation, I have a draft already as the build process was described for the earlier version, but was not upgraded since the HCL2 updates. The previous version's docs is available [here](https://github.com/gusztavvargadr/packer/tree/2310.0.0?tab=readme-ov-file#getting-started), and the overall process is the same (a .NET build "script" wrapping the Packer and Vagrant operations), but there were slight changes in the parameters. Anyhow, you can take a look at the concepts there.

I am building myself on Windows, Ubuntu and Mac hosts for various providers, so building for your use case should not be that complicated. Applying to other CI servers should be relatively easy as well, as you only need to orchestrate the existing build scripts which are supposed to support running anywhere.

I am planning to release a new version of the boxes soon and as part of that will update the description properly, will keep you posted. In the meantime, you can take a look at the current build jobs [here](https://github.com/gusztavvargadr/packer/blob/2404.0.0/.azure-pipelines/jobs.yml), which should enlist the steps needed to build native and vagrant images.

Please let me know.

**Reply by salderma** [reply]
Wonderful, will be looking forward to the new Docs!

After several hours of fiddling, I've been able to get the first native build running after finding the sort of order of steps from the older docs and applying the syntax from the current azure scripts.

**Reply by gusztavvargadr** [reply]
@salderma thanks for your efforts!

If you can list in a few bullet points what were the main pain points (besides the lack of up to date documentation), I'd make sure to include those in the docs explicitly, to help others in the future.

Thanks again!
